### PR TITLE
Switch to nginx buildpack and enable caching

### PR DESCRIPTION
**Pull request by 36degrees**
Unfortunately the staticfile buildpack has limitations which make it difficult to configure caching (you can only introduce additional nginx config within the location block) so this moves us on to the nginx buildpack which allows (and requires) us to provide the entire nginx configuration.

### CODE REVIEW

File: deploy/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  # Because images and videos are not fingerprinted we only cache them for 30
+  # minutes.
+  # 
+  # The epoch parameter corresponds to the absolute time
+  # 'Thu, 01 Jan 1970 00:00:01 GMT'.
+
+  map $sent_http_content_type $expires {
+    default                         off;
+    text/html                       epoch;
+    text/css                        max;
```
Comment (nickcolley) [inline deploy/nginx.conf]: I suppose new assets will have a different name but setting max expires makes me somewhat uneasy.

Comment (36degrees) [reply deploy/nginx.conf]: What do you suggest?

**Comment by 36degrees** [comment]
Should have said – this branch is currently deployed to https://govuk-design-system-caching-test.cloudapps.digital/

File: deploy/nginx.conf [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+  # Enabling sendfile eliminates copying file data into buffer and sends it
+  # directly
+
+  sendfile on;
+
+  # Enable gzip for text-based resources
+
+  gzip on;
+  gzip_disable "msie6";
```
Comment (LeePorte) [inline deploy/nginx.conf]: Do we need to support IE6?

File: deploy/nginx.conf [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    ~image/                         30m;
+    ~video/                         30m;
+  }
+
+  expires $expires;
+
+  # Set up a server!
+
+  server {
+    listen {{.Port}};
```
Comment (blairboy362) [inline deploy/nginx.conf]: You could add `default_server` to this line, then the `server_name` below effectively becomes optional

**Pull request merged**